warehouse/query-core/src/main/java/datawave/query/tables/keyword/StatefulKeywordUUIDChainStrategy.java

```diff
@@ -117,41 +128,91 @@ public String captureResultsAndBuildQuery(Iterator<Entry<Key,Value>> initialQuer
             Document document = documentEntry.getValue();
             final Map<String,Attribute<? extends Comparable<?>>> documentData = document.getDictionary();
 
-            List<String> identifiers = null;
-            List<String> languages = null;
+            String docId = row + "/" + dataType + "/" + uid;
 
-            for (Entry<String,Attribute<? extends Comparable<?>>> data : documentData.entrySet()) {
-                if (data.getKey().equals("LANGUAGE")) {
-                    languages = KeywordQueryUtil.getStringValuesFromAttribute(data.getValue());
-                } else if (data.getKey().equals("HIT_TERM")) {
-                    identifiers = KeywordQueryUtil.getStringValuesFromAttribute(data.getValue());
+            // apply all extractors to the document
```

Comment:
```suggestion
            // apply all extractors to the document data returned by the first stage of the query (e.g., lookupUUID)
            // in some cases this will be sufficient to generate input for the tag clouds we want.
```